zuul-core/src/main/java/com/netflix/zuul/netty/connectionpool/ConnectionPoolMetrics.java

```diff
@@ -0,0 +1,91 @@
+                                    Counter maxConnsPerHostExceededCounter,
+                                    Counter closeWrtBusyConnCounter,
+                                    Counter circuitBreakerClose,
+                                    PercentileTimer connEstablishTimer,
+                                    AtomicInteger connsInPool,
+                                    AtomicInteger connsInUse) {
+
+    public static final String METRIC_PREFIX = "connectionpool_";
+
+    public static ConnectionPoolMetrics create(OriginName originName, Registry registry) {
+        Counter createNewConnCounter = newCounter("create", originName, registry);
+        Counter createConnSucceededCounter = newCounter("create_success", originName, registry);
+        Counter createConnFailedCounter = newCounter("create_fail", originName, registry);
+
+        Counter closeConnCounter = newCounter("close", originName, registry);
+        Counter closeAbovePoolHighWaterMarkCounter = newCounter("closeAbovePoolHighWaterMark", originName, registry);
+        Counter closeExpiredConnLifetimeCounter = newCounter("closeExpiredConnLifetime", originName, registry);
+        Counter requestConnCounter = newCounter("request", originName, registry);
+        Counter reuseConnCounter = newCounter("reuse", originName, registry);
+        Counter releaseConnCounter = newCounter("release", originName, registry);
+        Counter alreadyClosedCounter = newCounter("alreadyClosed", originName, registry);
+        Counter connTakenFromPoolIsNotOpen = newCounter("fromPoolIsClosed", originName, registry);
+        Counter maxConnsPerHostExceededCounter = newCounter("maxConnsPerHostExceeded", originName, registry);
+        Counter closeWrtBusyConnCounter = newCounter("closeWrtBusyConnCounter", originName, registry);
+        Counter circuitBreakerClose = newCounter("closeCircuitBreaker", originName, registry);
+
+        PercentileTimer connEstablishTimer = PercentileTimer.get(
+                registry, registry.createId(METRIC_PREFIX + "createTiming", "id", originName.getMetricId()));
+
+        AtomicInteger connsInPool = newGauge("inPool", originName, registry);
+        AtomicInteger connsInUse = newGauge("inUse", originName, registry);
+
+        return new ConnectionPoolMetrics(createNewConnCounter, createConnSucceededCounter, createConnFailedCounter,
+                closeConnCounter, closeAbovePoolHighWaterMarkCounter, closeExpiredConnLifetimeCounter, requestConnCounter,
+                reuseConnCounter, releaseConnCounter, alreadyClosedCounter, connTakenFromPoolIsNotOpen,
+                maxConnsPerHostExceededCounter, closeWrtBusyConnCounter, circuitBreakerClose, connEstablishTimer, connsInPool, connsInUse);
+    }
+
+    private static Counter newCounter(String metricName, OriginName originName, Registry registry) {
+        return registry.counter(METRIC_PREFIX + metricName, "id", originName.getMetricId());
```

Comment:
Could you perhaps inline this into the `create` method instead, and remove the `METRIC_PREFIX +` concat? Normally when I'm searching around for where a metric is emitted, it's much easier to just match on the whole name, e.g. `connectionpool_create`, so if it's just all listed there it is a lot easier to locate

Reply:
for sure, I find the prefixes annoying as well